security/Member.php

```diff
@@ -1470,25 +1470,30 @@ public function fieldLabels($includerelations = true) {
 	 * This is likely to be customized for social sites etc. with a looser permission model.
 	 */
 	public function canView($member = null) {
-		if(!$member || !(is_a($member, 'Member')) || is_numeric($member)) $member = Member::currentUser();
-
-		// extended access checks
-		$results = $this->extend('canView', $member);
-		if($results && is_array($results)) {
-			if(!min($results)) return false;
-			else return true;
+		//get member
+		if(!($member instanceof Member)) {
+			$member = Member::currentUser();
+		}
+		//check for extensions, we do this first as they can overrule everything
+		$extended = $this->extendedCan(__FUNCTION__, $member);
+		if($extended !== null) {
+			return $extended;
 		}
+		
+		//need to be logged in and/or
+		//most checks below rely on $member being a Member
+		if(!$member) return false;		
 
-		// members can usually edit their own record
-		if($member && $this->ID == $member->ID) return true;
+		// members can usually view their own record
+		if($this->ID == $member->ID) return true;
 
+		//standard check
 		if(
```

Comment:
This whole block can be shortened to

```php
return Permission::checkMember($member, 'CMS_ACCESS_SecurityAdmin');
```

ADMIN permissions will return true for any security check, so don't need to check it explicitly.